File: src/pattern/adapter/java/Banner.java
package pattern.adapter.java;

public class Banner {
    String name;

    public Banner(String name){
        this.name = name;
    }

    public void showWithParen(){
        System.out.println("("+this.name+")");
    }

    public void showWithAster(){
        System.out.println("*"+this.name+"*");
    }
}

File: src/pattern/adapter/kotlin/Banner.kt
package pattern.adapter.kotlin

open class Banner(var name: String){
    var phone: String = "Default Phone Number"

    constructor(name: String, phone: String): this(name){
        var phone = phone
    }
    fun showWithParen(){
        println("($name)")
        println(phone)
    }

    fun showWithAster(){
        println("*$name*")
        println(phone)
    }
}